### DISCUSSION TITLE
Carbon Data Interchange

### DISCUSSION THREAD

**Discussion by seanmcilroy29** [Green-Software-Foundation/sci#334] [Ideas]
Submitted by Chris Lloyd-Jones

I've been chatting with people at Avanade, and we've noticed there are a lot of solutions tracking, producing and consuming carbon data right now, including:

**Tools that track carbon data**

- Cloud Carbon Footprint
- Microsoft Sustainability Manager
- SAP Carbon Footprint Analytics
- Avanade's Isotope - https://azuremarketplace.microsoft.com/en-us/marketplace/apps/avanade-5299580.amp_isotope_access?tab=overview

**Tools that produce carbon data**

- Carbon Aware SDK
- Metrikus - https://www.metrikus.io/
- CAST-IT - https://www.castsoftware.com/green-it
- Carbon CI Pipeline

**Tools that consume carbon data**

Cloverly https://www.cloverly.com/

I'm noticing a common problem:

- Tools don't measure discrete units of carbon in the same way, for example accounting for avoided carbon, or offset carbon
- Tools don't use the same measurement standards, for either co2eq, or the units of measurement
- It's difficult to link data from one system with another, without some kind of interchange or common format

**Suggestion**
Is there scope for a Carbon Data Interchange project within the Standard or Open Source Working group at GSF?

I propose the project would:

- Create a common data interchange format for sharing data about carbon between systems, ideally built upon an existing markup language like JSON-LD or YAML, so that a schema can be created

This would provide a format that tools and organizations can use to share carbon data, in a standardized way, to avoid re-inventing the wheel in exports and imports.

If interested, I'm happy to include folks from Avanade that have an interest in supporting this type of project, and I'm sure many of your organisations have the same.

**Distro**
I've included Open Source WG & Standards WG chairs, and folks who I think might be interested, based on the projects they work on, but please feel free to share more wid

**Comment by adrianco** [comment]
The Pathfinder project seems relevant
https://www.wbcsd.org/Programs/Climate-and-Energy/Climate/SOS-1.5/Resources/Pathfinder-Framework-Guidance-for-the-Accounting-and-Exchange-of-Product-Life-Cycle-Emissions

**Comment by atg-abhishek** [comment]
Based on the email discussion, the next step should be to develop a formal project proposal that would enough sufficient resources / detail to ensure success. cc @Sealjay

**Reply by Sealjay** [reply]
Agreed; as a way forward, the steering committee agreed to create a "Convention" project in the OSS WG, and then migrate this to the Standards WG if we decide to make this a formalised standard.

**Comment by Henry-WattTime** [comment]
Henry/Gadhu: Might overlap with new project in open source working group (Universal Software Carbon  Measurement (USCM) API  – PRF)
Srini: abstract out calculation methodology away from individual, provide raw emissions data from hardware/software configuration provided by user
Chris/Joshua at Avanade leading this project, Srini to reach out to coordinate overlap between USCM and this proposed potential project.

**Comment by srini1978** [comment]
@Henry-WattTime  Going through the specifics of Carbon Data interchange project , it is focused more on identifying a standard format for exchange of carbon data between systems . As @Sealjay  has stated above, the intent is  to "Create a common data interchange format for sharing data about carbon between systems, ideally built upon an existing markup language like JSON-LD or YAML".

However the scope of Carbon QL or Universal Software Carbon Measurement API is to provide the value for each of the constituent items of the SCI formula - E, I, M and R by tapping into existing Carbon datasets. The output of this API will be a MtCo2eq value for E, I, M depending upon the "Software boundary" that the user will be giving as input.  Behind the facade, the USCM project will tap into the standard datasets maintained by the SCI open data project.

Hence these 2 projects have completely different ambitions, according to me.  @Sealjay  @jawache  @navveenb  Feel free to add your thoughts.

**Reply by Sealjay** [reply]
Hi folks, I agree, the ambition is different for CarbonQL and CDIF; although there are some dependencies I see between the two, which we can hopefully resolve within the open source working group when they crop up; @srini1978 will be leading the CarbonQL project, and Josh (will find his GitHub ID) will be leading the CDIF project.

We discussed in the working group call last Wednesday - but by dependencies, I would expect the CarbonQL / USCM API would ingest data that is in the CDI format, and output data in the same format.

**Reply by Sealjay** [reply]
Tagging Josh - @KnowledgeRatio

**Comment by Henry-WattTime** [comment]
WG: Looking forward to seeing and reviewing the project proposal. Project may sit in the open source or standards WG.

**Reply by Sealjay** [reply]
Thanks. Over to you @KnowledgeRatio

**Comment by Sealjay** [comment]
Suggest we close this discussion now we have a way forward, with Carbon QL + the Real time Data standards project.
